src/Kliva/Models/ActivitySummaryCacheLoader.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.UI.Xaml.Data;
+using DynamicData;
+using Kliva.Services;
+
+namespace Kliva.Models
+{
+    internal class ActivitySummaryCacheLoader
+    {
+        public delegate Task<string> LoadMethod(int page, int pageSize);
+
+        private readonly LoadMethod _loader;
+        private readonly Func<string, Task<List<ActivitySummary>>> _hydrater;
+        private readonly int _pageSize;
+        private readonly ISourceCache<ActivitySummary, long> _targetCache;
+        private readonly string _backingStoreCacheName;
+        private int _page;
+        private bool _suppressNextBackingCacheGet = false;
+
+        public ActivitySummaryCacheLoader(
+            LoadMethod loader,
+            Func<string, Task<List<ActivitySummary>>> hydrater,
+            int pageSize, ISourceCache<ActivitySummary, long> targetCache,
+            string backingStoreCacheName)
+        {
+            if (loader == null) throw new ArgumentNullException(nameof(loader));
+            if (hydrater == null) throw new ArgumentNullException(nameof(hydrater));
+            if (targetCache == null) throw new ArgumentNullException(nameof(targetCache));
+            if (backingStoreCacheName == null) throw new ArgumentNullException(nameof(backingStoreCacheName));
+
+            _loader = loader;
+            _hydrater = hydrater;
+            _pageSize = pageSize;            
+            _targetCache = targetCache;
+            _backingStoreCacheName = backingStoreCacheName;
+        }
+
+        public IAsyncOperation<LoadMoreItemsResult> LoadMoreItems(uint count)
+        {
+            return AsyncInfo.Run(_ => LoadMoreItemsAsync());
+        }
+
+        public async void Reset()
+        {
+            _page = 0;
+            _targetCache.Clear();
+            _suppressNextBackingCacheGet = true;
+            await LoadMoreItemsAsync();
+        }
+
+        public bool IsLoading { get; private set; }        
+
+        private async Task<LoadMoreItemsResult> LoadMoreItemsAsync()
+        {
```

Comment:
Not sure if this is the culprit or not... but difference in data loading with current version versus this code, leads to blocking UI.
Examples, you can't close the what's new popup while the list is doing it's initial load. ( not sure what code part this applies too? )
Other example, if you scroll down do do more loading, you can't move the list until the loading is done. ( hence why I think this is the code with the problem )

The read and write methods should be pushed back to another thread. I would suggest trying to use Task.Run to see if this would make things better.

@bartlannoeye do you see the same blocking issues?